Manual build with CMake fails

### Describe the bug

manual build with CMake using clang complier after executing `ninja` fails.

```
clang++ --version
Ubuntu clang version 17.0.6 (++20231209124227+6009708b4367-1~exp1~20231209124336.77)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

the long error: 
`ninja
[0/2] Re-checking globbed directories...
[2/256] Building CXX object tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/eth_tunneler.cpp.o
FAILED: tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/eth_tunneler.cpp.o 
/usr/bin/clang++-17 -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DFMT_HEADER_ONLY=1 -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_TRACE -DSPDLOG_FMT_EXTERNAL -DSPDLOG_FWRITE_UNLOCKED -DTT_ENABLE_LIGHT_METAL_TRACE=1 -I/home/ubuntu/ionnx/tt-metal/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/api/tt-metalium -I/home/ubuntu/ionnx/tt-metal/build/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/debug -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/profiler -I/home/ubuntu/ionnx/tt-metal/tt_metal/include -I/home/ubuntu/ionnx/tt-metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/common -I/home/ubuntu/ionnx/tt-metal/tt_stl/. -I/home/ubuntu/ionnx/tt-metal/tt_metal/hostdevcommon/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw/inc -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw -I/home/ubuntu/ionnx/tt-metal/tt_metal/llrt -isystem /home/ubuntu/ionnx/tt-metal/build/tt_metal/impl/flatbuffers -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/nlohmann_json/798e0374658476027d9723eeb67a262d0f3c8308/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/magic_enum/4d76fe0a5b27a0e62d6c15976d02b33c54207096/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/fmt/69912fb6b71fcb1f7e5deca191a2bb4748c4e7b6/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/core/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/config/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/static_assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/throw_exception/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/container/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/intrusive/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/move/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/reflect/f93e77475670eaeacf332927dfe8b50e3f3812e0 -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/tt-logger/37d08f9c0e2a5cc0247e0cf050c06563597d9c7c/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/spdlog/b1c2586bb5c35a7929362e87f62433eb68206873/include -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/src/firmware/riscv -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/device/api -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/simd-everywhere/b3b426f78574ef837b17f42e86bab88314c5e4db -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/tracy/public -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/smart_ptr/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/type_traits/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/flatbuffers/2c4062bffa52fa4157b1b4deeae73395df475fda/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/range-v3/0_12_0_patched/include -std=c++20 -fPIC -fsized-deallocation -mavx2 -fPIC -fvisibility-inlines-hidden -Wall -Werror -Wconditional-uninitialized -Wno-deprecated-declarations -Wno-unused-function -Wno-unused-variable -Wno-c++11-narrowing -Wno-deprecated-volatile -Wno-for-loop-analysis -Wno-logical-op-parentheses -Wno-mismatched-tags -Wno-reorder-ctor -Wno-unused-lambda-capture -Wno-unused-private-field -Wno-unknown-pragmas -Xclang -fno-pch-timestamp -Wno-int-to-pointer-cast -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -MD -MT tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/eth_tunneler.cpp.o -MF tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/eth_tunneler.cpp.o.d -o tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/eth_tunneler.cpp.o -c /home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/eth_tunneler.cpp
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/eth_tunneler.cpp:76:28: error: variable 'tunneler_kernel' set but not used [-Werror,-Wunused-but-set-variable]
   76 |         EthTunnelerKernel* tunneler_kernel = nullptr;
      |                            ^
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/eth_tunneler.cpp:204:28: error: variable 'ds_tunneler_kernel' set but not used [-Werror,-Wunused-but-set-variable]
  204 |         EthTunnelerKernel* ds_tunneler_kernel = nullptr;
      |                            ^
2 errors generated.
[3/256] Building CXX object tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/dispatch.cpp.o
FAILED: tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/dispatch.cpp.o 
/usr/bin/clang++-17 -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DFMT_HEADER_ONLY=1 -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_TRACE -DSPDLOG_FMT_EXTERNAL -DSPDLOG_FWRITE_UNLOCKED -DTT_ENABLE_LIGHT_METAL_TRACE=1 -I/home/ubuntu/ionnx/tt-metal/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/api/tt-metalium -I/home/ubuntu/ionnx/tt-metal/build/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/debug -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/profiler -I/home/ubuntu/ionnx/tt-metal/tt_metal/include -I/home/ubuntu/ionnx/tt-metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/common -I/home/ubuntu/ionnx/tt-metal/tt_stl/. -I/home/ubuntu/ionnx/tt-metal/tt_metal/hostdevcommon/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw/inc -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw -I/home/ubuntu/ionnx/tt-metal/tt_metal/llrt -isystem /home/ubuntu/ionnx/tt-metal/build/tt_metal/impl/flatbuffers -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/nlohmann_json/798e0374658476027d9723eeb67a262d0f3c8308/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/magic_enum/4d76fe0a5b27a0e62d6c15976d02b33c54207096/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/fmt/69912fb6b71fcb1f7e5deca191a2bb4748c4e7b6/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/core/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/config/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/static_assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/throw_exception/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/container/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/intrusive/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/move/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/reflect/f93e77475670eaeacf332927dfe8b50e3f3812e0 -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/tt-logger/37d08f9c0e2a5cc0247e0cf050c06563597d9c7c/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/spdlog/b1c2586bb5c35a7929362e87f62433eb68206873/include -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/src/firmware/riscv -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/device/api -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/simd-everywhere/b3b426f78574ef837b17f42e86bab88314c5e4db -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/tracy/public -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/smart_ptr/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/type_traits/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/flatbuffers/2c4062bffa52fa4157b1b4deeae73395df475fda/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/range-v3/0_12_0_patched/include -std=c++20 -fPIC -fsized-deallocation -mavx2 -fPIC -fvisibility-inlines-hidden -Wall -Werror -Wconditional-uninitialized -Wno-deprecated-declarations -Wno-unused-function -Wno-unused-variable -Wno-c++11-narrowing -Wno-deprecated-volatile -Wno-for-loop-analysis -Wno-logical-op-parentheses -Wno-mismatched-tags -Wno-reorder-ctor -Wno-unused-lambda-capture -Wno-unused-private-field -Wno-unknown-pragmas -Xclang -fno-pch-timestamp -Wno-int-to-pointer-cast -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -MD -MT tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/dispatch.cpp.o -MF tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/dispatch.cpp.o.d -o tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/dispatch.cpp.o -c /home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/dispatch.cpp
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/dispatch.cpp:294:14: error: variable 'found_prefetch_h' set but not used [-Werror,-Wunused-but-set-variable]
  294 |         bool found_prefetch_h = false;
      |              ^
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/dispatch.cpp:295:14: error: variable 'found_relay_mux' set but not used [-Werror,-Wunused-but-set-variable]
  295 |         bool found_relay_mux = false;
      |              ^
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/dispatch.cpp:360:14: error: variable 'found_relay_mux' set but not used [-Werror,-Wunused-but-set-variable]
  360 |         bool found_relay_mux = false;  // fabric mux
      |              ^
3 errors generated.
[7/256] Building CXX object tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/prefetch.cpp.o
FAILED: tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/prefetch.cpp.o 
/usr/bin/clang++-17 -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DFMT_HEADER_ONLY=1 -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_TRACE -DSPDLOG_FMT_EXTERNAL -DSPDLOG_FWRITE_UNLOCKED -DTT_ENABLE_LIGHT_METAL_TRACE=1 -I/home/ubuntu/ionnx/tt-metal/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/api/tt-metalium -I/home/ubuntu/ionnx/tt-metal/build/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/debug -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/profiler -I/home/ubuntu/ionnx/tt-metal/tt_metal/include -I/home/ubuntu/ionnx/tt-metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/common -I/home/ubuntu/ionnx/tt-metal/tt_stl/. -I/home/ubuntu/ionnx/tt-metal/tt_metal/hostdevcommon/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw/inc -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw -I/home/ubuntu/ionnx/tt-metal/tt_metal/llrt -isystem /home/ubuntu/ionnx/tt-metal/build/tt_metal/impl/flatbuffers -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/nlohmann_json/798e0374658476027d9723eeb67a262d0f3c8308/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/magic_enum/4d76fe0a5b27a0e62d6c15976d02b33c54207096/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/fmt/69912fb6b71fcb1f7e5deca191a2bb4748c4e7b6/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/core/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/config/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/static_assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/throw_exception/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/container/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/intrusive/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/move/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/reflect/f93e77475670eaeacf332927dfe8b50e3f3812e0 -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/tt-logger/37d08f9c0e2a5cc0247e0cf050c06563597d9c7c/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/spdlog/b1c2586bb5c35a7929362e87f62433eb68206873/include -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/src/firmware/riscv -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/device/api -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/simd-everywhere/b3b426f78574ef837b17f42e86bab88314c5e4db -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/tracy/public -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/smart_ptr/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/type_traits/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/flatbuffers/2c4062bffa52fa4157b1b4deeae73395df475fda/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/range-v3/0_12_0_patched/include -std=c++20 -fPIC -fsized-deallocation -mavx2 -fPIC -fvisibility-inlines-hidden -Wall -Werror -Wconditional-uninitialized -Wno-deprecated-declarations -Wno-unused-function -Wno-unused-variable -Wno-c++11-narrowing -Wno-deprecated-volatile -Wno-for-loop-analysis -Wno-logical-op-parentheses -Wno-mismatched-tags -Wno-reorder-ctor -Wno-unused-lambda-capture -Wno-unused-private-field -Wno-unknown-pragmas -Xclang -fno-pch-timestamp -Wno-int-to-pointer-cast -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -MD -MT tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/prefetch.cpp.o -MF tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/prefetch.cpp.o.d -o tt_metal/impl/CMakeFiles/impl.dir/dispatch/kernel_config/prefetch.cpp.o -c /home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/prefetch.cpp
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/prefetch.cpp:249:14: error: variable 'found_dispatch' set but not used [-Werror,-Wunused-but-set-variable]
  249 |         bool found_dispatch = false;
      |              ^
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/prefetch.cpp:250:14: error: variable 'found_dispatch_s' set but not used [-Werror,-Wunused-but-set-variable]
  250 |         bool found_dispatch_s = false;
      |              ^
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/prefetch.cpp:363:14: error: variable 'found_dispatch' set but not used [-Werror,-Wunused-but-set-variable]
  363 |         bool found_dispatch = false;
      |              ^
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/prefetch.cpp:364:14: error: variable 'found_dispatch_s' set but not used [-Werror,-Wunused-but-set-variable]
  364 |         bool found_dispatch_s = false;
      |              ^
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/kernel_config/prefetch.cpp:365:14: error: variable 'found_relay_mux' set but not used [-Werror,-Wunused-but-set-variable]
  365 |         bool found_relay_mux = false;
      |              ^
5 errors generated.
[10/256] Building CXX object tt_metal/impl/CMakeFiles/impl.dir/dispatch/topology.cpp.o
FAILED: tt_metal/impl/CMakeFiles/impl.dir/dispatch/topology.cpp.o 
/usr/bin/clang++-17 -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DFMT_HEADER_ONLY=1 -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_TRACE -DSPDLOG_FMT_EXTERNAL -DSPDLOG_FWRITE_UNLOCKED -DTT_ENABLE_LIGHT_METAL_TRACE=1 -I/home/ubuntu/ionnx/tt-metal/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/api/tt-metalium -I/home/ubuntu/ionnx/tt-metal/build/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/debug -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/profiler -I/home/ubuntu/ionnx/tt-metal/tt_metal/include -I/home/ubuntu/ionnx/tt-metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/common -I/home/ubuntu/ionnx/tt-metal/tt_stl/. -I/home/ubuntu/ionnx/tt-metal/tt_metal/hostdevcommon/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw/inc -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw -I/home/ubuntu/ionnx/tt-metal/tt_metal/llrt -isystem /home/ubuntu/ionnx/tt-metal/build/tt_metal/impl/flatbuffers -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/nlohmann_json/798e0374658476027d9723eeb67a262d0f3c8308/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/magic_enum/4d76fe0a5b27a0e62d6c15976d02b33c54207096/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/fmt/69912fb6b71fcb1f7e5deca191a2bb4748c4e7b6/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/core/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/config/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/static_assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/throw_exception/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/container/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/intrusive/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/move/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/reflect/f93e77475670eaeacf332927dfe8b50e3f3812e0 -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/tt-logger/37d08f9c0e2a5cc0247e0cf050c06563597d9c7c/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/spdlog/b1c2586bb5c35a7929362e87f62433eb68206873/include -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/src/firmware/riscv -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/device/api -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/simd-everywhere/b3b426f78574ef837b17f42e86bab88314c5e4db -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/tracy/public -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/smart_ptr/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/type_traits/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/flatbuffers/2c4062bffa52fa4157b1b4deeae73395df475fda/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/range-v3/0_12_0_patched/include -std=c++20 -fPIC -fsized-deallocation -mavx2 -fPIC -fvisibility-inlines-hidden -Wall -Werror -Wconditional-uninitialized -Wno-deprecated-declarations -Wno-unused-function -Wno-unused-variable -Wno-c++11-narrowing -Wno-deprecated-volatile -Wno-for-loop-analysis -Wno-logical-op-parentheses -Wno-mismatched-tags -Wno-reorder-ctor -Wno-unused-lambda-capture -Wno-unused-private-field -Wno-unknown-pragmas -Xclang -fno-pch-timestamp -Wno-int-to-pointer-cast -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -MD -MT tt_metal/impl/CMakeFiles/impl.dir/dispatch/topology.cpp.o -MF tt_metal/impl/CMakeFiles/impl.dir/dispatch/topology.cpp.o.d -o tt_metal/impl/CMakeFiles/impl.dir/dispatch/topology.cpp.o -c /home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/topology.cpp
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/dispatch/topology.cpp:872:22: error: variable 'found_remote' set but not used [-Werror,-Wunused-but-set-variable]
  872 |                 bool found_remote = false;
      |                      ^
1 error generated.
[16/256] Building CXX object tt_metal/impl/CMakeFiles/impl.dir/profiler/profiler.cpp.o
FAILED: tt_metal/impl/CMakeFiles/impl.dir/profiler/profiler.cpp.o 
/usr/bin/clang++-17 -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DFMT_HEADER_ONLY=1 -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_TRACE -DSPDLOG_FMT_EXTERNAL -DSPDLOG_FWRITE_UNLOCKED -DTT_ENABLE_LIGHT_METAL_TRACE=1 -I/home/ubuntu/ionnx/tt-metal/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/api/tt-metalium -I/home/ubuntu/ionnx/tt-metal/build/tt_metal/api -I/home/ubuntu/ionnx/tt-metal/tt_metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/debug -I/home/ubuntu/ionnx/tt-metal/tt_metal/impl/profiler -I/home/ubuntu/ionnx/tt-metal/tt_metal/include -I/home/ubuntu/ionnx/tt-metal -I/home/ubuntu/ionnx/tt-metal/tt_metal/common -I/home/ubuntu/ionnx/tt-metal/tt_stl/. -I/home/ubuntu/ionnx/tt-metal/tt_metal/hostdevcommon/api -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw/inc -I/home/ubuntu/ionnx/tt-metal/tt_metal/hw -I/home/ubuntu/ionnx/tt-metal/tt_metal/llrt -isystem /home/ubuntu/ionnx/tt-metal/build/tt_metal/impl/flatbuffers -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/nlohmann_json/798e0374658476027d9723eeb67a262d0f3c8308/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/magic_enum/4d76fe0a5b27a0e62d6c15976d02b33c54207096/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/fmt/69912fb6b71fcb1f7e5deca191a2bb4748c4e7b6/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/core/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/config/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/static_assert/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/throw_exception/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/container/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/intrusive/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/move/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/reflect/f93e77475670eaeacf332927dfe8b50e3f3812e0 -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/tt-logger/37d08f9c0e2a5cc0247e0cf050c06563597d9c7c/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/spdlog/b1c2586bb5c35a7929362e87f62433eb68206873/include -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/src/firmware/riscv -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/umd/device/api -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/simd-everywhere/b3b426f78574ef837b17f42e86bab88314c5e4db -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -isystem /home/ubuntu/ionnx/tt-metal/tt_metal/third_party/tracy/public -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/smart_ptr/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/type_traits/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/flatbuffers/2c4062bffa52fa4157b1b4deeae73395df475fda/include -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/range-v3/0_12_0_patched/include -std=c++20 -fPIC -fsized-deallocation -mavx2 -fPIC -fvisibility-inlines-hidden -Wall -Werror -Wconditional-uninitialized -Wno-deprecated-declarations -Wno-unused-function -Wno-unused-variable -Wno-c++11-narrowing -Wno-deprecated-volatile -Wno-for-loop-analysis -Wno-logical-op-parentheses -Wno-mismatched-tags -Wno-reorder-ctor -Wno-unused-lambda-capture -Wno-unused-private-field -Wno-unknown-pragmas -Xclang -fno-pch-timestamp -Wno-int-to-pointer-cast -isystem /home/ubuntu/ionnx/tt-metal/.cpmcache/taskflow/52063f60902bfeb362fa4616b1394ab5efe30994 -MD -MT tt_metal/impl/CMakeFiles/impl.dir/profiler/profiler.cpp.o -MF tt_metal/impl/CMakeFiles/impl.dir/profiler/profiler.cpp.o.d -o tt_metal/impl/CMakeFiles/impl.dir/profiler/profiler.cpp.o -c /home/ubuntu/ionnx/tt-metal/tt_metal/impl/profiler/profiler.cpp
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/profiler/profiler.cpp:326:22: error: variable 'riscNumRead' set but not used [-Werror,-Wunused-but-set-variable]
  326 |             uint32_t riscNumRead = 0;
      |                      ^
/home/ubuntu/ionnx/tt-metal/tt_metal/impl/profiler/profiler.cpp:327:22: error: variable 'coreFlatIDRead' set but not used [-Werror,-Wunused-but-set-variable]
  327 |             uint32_t coreFlatIDRead = 0;
      |                      ^
2 errors generated.
[17/256] Building CXX object tt_metal/impl/CMakeFiles/impl.dir/trace/trace.cpp.o
ninja: build stopped: subcommand failed.
`

### Steps to reproduce the issue

1. Go to 'main branch - commit: 308bc60537c8032a36fc5b03147bd3c5b0f30acb'
2. following the install instructions option B from [metal install page.: Option B: Manual Build with CMake](https://github.com/tenstorrent/tt-metal/blob/main/INSTALLING.md#source) run: 
`mkdir build`
`cd build`
`cmake .. -G Ninja -DCMAKE_CXX_COMPILER=clang++-17 -DCMAKE_BUILD_TYPE=RelWithDebugInfo`
`ninja`

3. After running `ninja` command, see the error. 

![Image](https://github.com/user-attachments/assets/185db104-350a-4b70-bf3c-24b74a687263)


### Expected behavior

to be able to build manually without errors. 

### Please complete the following environment information

- OS: 22.04
- python version: Python 3.10.12
- branch main - commit: `308bc60537c8032a36fc5b03147bd3c5b0f30acb`

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -165,7 +165,7 @@ add_compile_options(
     -Wno-deprecated-declarations
     # Vars that are used only in Asserts will appear as dead stores in builds
     # where the preprocessor has stripped away the assert.
-    "$<$<OR:$<CONFIG:Release>,$<CXX_COMPILER_ID:GNU>>:-Wno-unused-but-set-variable>"
+    -Wno-unused-but-set-variable
     -Wno-unused-function
     -Wno-unused-variable
     "$<$<CXX_COMPILER_ID:Clang>:-Wno-c++11-narrowing>"

